mempool/mempool_test.go

```diff
@@ -866,3 +930,853 @@ func TestCheckSpend(t *testing.T) {
+				if err != nil {
+					ctx.t.Fatalf("unable to create "+
+						"transaction: %v", err)
+				}
+
+				return tx, nil
+			},
+		},
+		{
+			name: "conflicts",
+			setup: func(ctx *testContext) (*btcutil.Tx, []*btcutil.Tx) {
+				coinbase1 := ctx.addCoinbaseTx(1)
+				coinbaseOut1 := txOutToSpendableOut(coinbase1, 0)
+				outs := []spendableOutput{coinbaseOut1}
+				conflict1 := ctx.addSignedTx(
+					outs, 1, 0, false, false,
+				)
+
+				coinbase2 := ctx.addCoinbaseTx(1)
+				coinbaseOut2 := txOutToSpendableOut(coinbase2, 0)
+				outs = []spendableOutput{coinbaseOut2}
+				conflict2 := ctx.addSignedTx(
+					outs, 1, 0, false, false,
+				)
+
+				// Create a transaction that spends both
+				// coinbase outputs that were spent above.
+				outs = []spendableOutput{coinbaseOut1, coinbaseOut2}
+				tx, err := ctx.harness.CreateSignedTx(
+					outs, 1, 0, false,
+				)
+				if err != nil {
+					ctx.t.Fatalf("unable to create "+
+						"transaction: %v", err)
+				}
+
+				return tx, []*btcutil.Tx{conflict1, conflict2}
+			},
+		},
+		{
+			name: "descendant conflicts",
+			setup: func(ctx *testContext) (*btcutil.Tx, []*btcutil.Tx) {
+				coinbase := ctx.addCoinbaseTx(1)
+
+				// Create a replaceable parent that spends the
+				// coinbase output.
+				coinbaseOut := txOutToSpendableOut(coinbase, 0)
+				outs := []spendableOutput{coinbaseOut}
+				parent := ctx.addSignedTx(outs, 1, 0, false, false)
+
+				parentOut := txOutToSpendableOut(parent, 0)
+				outs = []spendableOutput{parentOut}
+				child := ctx.addSignedTx(outs, 1, 0, false, false)
+
+				// Create another transaction that spends the
+				// same coinbase output. Since the original
+				// spender of this output, all of its spends
+				// should also be conflicts.
+				outs = []spendableOutput{coinbaseOut}
+				tx, err := ctx.harness.CreateSignedTx(
+					outs, 2, 0, false,
+				)
+				if err != nil {
+					ctx.t.Fatalf("unable to create "+
+						"transaction: %v", err)
+				}
+
+				return tx, []*btcutil.Tx{parent, child}
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		success := t.Run(testCase.name, func(t *testing.T) {
+			// We'll start each test by creating our mempool
+			// harness.
+			harness, _, err := newPoolHarness(&chaincfg.MainNetParams)
+			if err != nil {
+				t.Fatalf("unable to create test pool: %v", err)
+			}
+			ctx := &testContext{t, harness}
+
+			// Each test includes a setup method, which will set up
+			// its required dependencies. The transaction returned
+			// is the one we'll be querying for the expected
+			// conflicts.
+			tx, conflicts := testCase.setup(ctx)
+
+			// Assert the expected conflicts are returned.
+			txConflicts := ctx.harness.txPool.txConflicts(tx)
+			if len(txConflicts) != len(conflicts) {
+				ctx.t.Fatalf("expected %d conflicts, got %d",
+					len(conflicts), len(txConflicts))
+			}
+			for _, conflict := range conflicts {
+				conflictHash := *conflict.Hash()
+				if _, ok := txConflicts[conflictHash]; !ok {
```

Comment:
This is not a totally accurate assertion, since technically `conflicts` could have duplicates.

Reply:
It's assumed that `conflicts` won't include duplicates since the `txConflicts` method doesn't as well.